lib/pbench/cli/server/report.py

```diff
@@ -56,18 +92,38 @@ class Watch:
     """
 
     def __init__(self, interval: float):
-        self.interval = datetime.timedelta(seconds=interval) if interval else None
-        self.start = datetime.datetime.now()
-        self.last = self.start
+        """Initialize the object.
+
+        Args:
+            interval: interval in seconds for status updates
+        """
+        self.start = time.time()
+        self.interval = interval
+        self.status = "starting"
+        if interval:
+            self.thread = Thread(target=self.watcher)
+            self.thread.setDaemon(True)
+            self.thread.start()
 
     def update(self, status: str):
-        now = datetime.datetime.now()
-        if self.interval and now > self.last + self.interval:
-            self.last = now
-            delta = now - self.start
-            hours, remainder = divmod(delta.seconds, 3600)
+        """Update status if appropriate.
+
+        If periodic updates are enabled, and {interval} time has elapsed since
+        the last update, print the update message.
+
+        Args:
+            status: status string
+        """
+        self.status = status
```

Comment:
So, you figure this operation is atomic enough that we don't need synchronization?  (I imagine that we don't care too much about which string gets printed, but it would be annoying if the lack of synchronization resulted in a segfault because the watcher's stale value referenced deallocated (or reallocated) memory.)

Reply:
I can't see how it matters, unless Python loses track of its live memory pointers: either we get the new message or the old message.

Reply:
So, if the one thread has a reference to the new message and the other thread has a stale reference to the old message, then the old message won't get garbage-collected?  OK...I'll believe that logic.